src/main/app/claims/featurePermissionMiddleware.ts

```diff
@@ -0,0 +1,25 @@
+import * as express from 'express'
+
+import { User } from 'idam/user'
+import { ClaimStoreClient } from 'claims/claimStoreClient'
+import { Paths as ClaimPaths } from 'claim/paths'
+import * as toBoolean from 'to-boolean'
+import * as config from 'config'
+
+export class FeaturePermissionMiddleware {
+
+  static retrieveUserConsentRole (req: express.Request, res: express.Response, next: express.NextFunction): void {
+    if (!toBoolean(config.get<boolean>('featureToggles.featuresPermission'))) {
+      return next()
+    }
+    const claimStoreClient: ClaimStoreClient = new ClaimStoreClient()
+    const user: User = res.locals.user
+    claimStoreClient.retrieveUserRoles(user).then(value => {
```

Comment:
You need to check for either of the specific roles being present, checking length isn't enough because there could be other roles in there